protocols/Leafcutter.sh

```diff
@@ -27,62 +31,44 @@ conditionCount=$(tail -n +2 "${projectJobsDir}/${project}.csv" | cut -d "," -f "
 where (0 = unstranded, 1 = first-strand/RF, 2, = second-strand/FR)."
 
-rm -f "${intermediateDir}${project}_juncfiles.txt"
-cd "${intermediateDir}"
-for bamfile in $(ls *.${sampleMergedBamExt}); do
-
-    echo Converting "${bamfile}" to "${bamfile}".junc
-    samtools index "${bamfile}"
-
-    #BUG: set to standed 0.
-    regtools junctions extract \
-    -a 8 \
-    -m 50 \
-    -M 500000 \
-    -s 0 \
-    "${bamfile}" \
-    -o "${bamfile}".junc
-
-    echo "${intermediateDir}${bamfile}".junc >> "${intermediateDir}${project}"_juncfiles.txt
-done
-
-python "${EBROOTLEAFCUTTER}"/clustering/leafcutter_cluster_regtools.py \
--j "${intermediateDir}/${project}"_juncfiles.txt \
--m 50 \
--r "${intermediateDir}" \
--o "${project}"_leafcutter_cluster_regtools \
--l 500000 \
---checkchrom
-
 echo "create group_list"
-awk -F',' '{print $1".sorted.merged.bam\t"$2}' "${intermediateDir}/metadata.csv" \
+col=$(col="externalSampleID"; head -n1 "${projectJobsDir}/${project}.csv" | tr "," "\n" | grep -n $col)
+colID=(${col//:/ })
+awk -F',' -v id=${colID[0]} -v con=${colArray[0]} '{print $id".sorted.merged.bam\t"$con}' "${projectJobsDir}/${project}.csv" \
 > "${intermediateDir}${project}_groups_file.txt"
 
 sed 1d "${intermediateDir}${project}_groups_file.txt" > "${intermediateDir}${project}"_groups_file.txt.tmp
 mv "${intermediateDir}${project}_groups_file.txt.tmp" "${intermediateDir}${project}_groups_file.txt"
 
+echo "conditionCount = ${conditionCount}"
 if [[ "${conditionCount}" -gt 1 ]]
 then
 	echo "Differential Splicing with ${conditionCount} groups."
 	Rscript "${EBROOTLEAFCUTTER}/scripts/leafcutter_ds.R" \
 	--num_threads 4 \
-	-o "${intermediateDir}${project}_leafcutter_ds" \
+        -i 1 \
+        -g 1 \
+        -c 3 \
+	-e "${annotationTxt}" \
+	-o "${tmpintermediateDir}${project}_leafcutter_ds" \
 	"${intermediateDir}${project}_leafcutter_cluster_regtools_perind_numers.counts.gz" \
 	"${intermediateDir}${project}_groups_file.txt"
 
 	Rscript "${EBROOTLEAFCUTTER}/scripts/ds_plots.R" \
 	-e "${EBROOTLEAFCUTTER}/annotation_codes/gencode_hg19/gencode_hg19_all_exons.txt.gz" \
-	-o "${intermediateDir}${project}_leafcutter_ds" \
+	-o "${tmpintermediateDir}${project}_leafcutter_ds" \
 	"${intermediateDir}${project}_leafcutter_cluster_regtools_perind_numers.counts.gz" \
 	"${intermediateDir}${project}_groups_file.txt" \
-	"${intermediateDir}${project}_leafcutter_ds_cluster_significance.txt" \
+	"${tmpintermediateDir}${project}_leafcutter_ds_cluster_significance.txt" \
 	-f 0.05
 
+	mv "${tmpintermediateDir}"/"${project}"* "${intermediateDir}"
 else
-       echo "Outlier Splicing, $NUMBERCONDITIONS conditions found."
+       echo "Outlier Splicing, $conditionCount conditions found."
 	Rscript	"${EBROOTLEAFCUTTER}/scripts/leafcutterMD.R" \
 	--num_threads 8 \
+	-o "${tmpintermediateDir}${project}" \
 	"${intermediateDir}${project}_leafcutter_cluster_regtools_perind_numers.counts.gz"
-fi 
 
-cd -
+	mv "${tmpintermediateDir}${project}"* "${intermediateDir}"
```

Comment:
to be a 100% sure, i would add a slash between ${tmpintermediateDir} and ${project}